server/routes/companies.js

```diff
@@ -2,14 +2,145 @@
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all company names
+router.get('/names', function(req, res, next) {
+  db.query('SELECT id, name FROM company ORDER BY name;', [], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all company names and logos
+router.get('/logos', function(req, res, next) {
+  db.query('SELECT id, name, logo FROM company ORDER BY name;', [], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all company names, logos, and websites
+router.get('/websites', function(req, res, next) {
+  db.query('SELECT id, name, logo, website FROM company ORDER BY name;', [], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET company info by company_id
+router.get('/:company_id', function(req, res, next) {
+  const company_id = req.params.company_id;
+  db.query('SELECT * FROM company WHERE id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all events associated with a given company
+router.get('/:company_id/events', function(req, res, next) {
+  const company_id = req.params.company_id;
+  db.query('SELECT event_id FROM event_company WHERE company_id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all positions a given company is hiring
+router.get('/:company_id/positions', function(req, res, next) {
+    const company_id = req.params.company_id;
+    db.query('SELECT * FROM company_position WHERE company_id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all majors a given company is hiring
+router.get('/:company_id/majors', function(req, res, next) {
+    const company_id = req.params.company_id;
+    db.query('SELECT * FROM company_major WHERE company_id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all contacts for a given company
+router.get('/:company_id/contacts', function(req, res, next) {
+    const company_id = req.params.company_id;
+    db.query('SELECT * FROM company_contact WHERE company_id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all users interested in a certain company
+router.get('/:company_id/users', function(req, res, next) {
+    const company_id = req.params.company_id;
+    db.query('SELECT user_id FROM user_company_rank WHERE company_id = $1;', [company_id], (err, result) => {
+    if (err) return next(err);
+    res.send(result.rows);
+  });
+});
+
+// GET all companies containing term substring in their name
+router.get('/search', function(req,res,next) {
+  const search = req.query.term;
+  if(search == undefined){
```

Comment:
Add space between if and (search... and between undefined) and {

Reply:
done!